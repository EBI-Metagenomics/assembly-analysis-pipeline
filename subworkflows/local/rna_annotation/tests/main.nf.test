nextflow_workflow {

    name "Test Subworkflow RNA_ANNOTATION"
    script "../main.nf"
    workflow "RNA_ANNOTATION"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_ebimetagenomics"
    tag "subworkflows/rna_annotation"
    tag "subworkflows/detect_rna"
    tag "easel"
    tag "easel/eslsfetch"
    tag "extractcoords"

    test("test_rna_annotation") {

        when {
            params {
                rfam_cm               = "${moduleDir}/tests/rfam/*.cm"
                rfam_claninfo         = "${moduleDir}/tests/ribo.claninfo"
                bgc_contigs_chunksize = '50M'
            }
            workflow {
                """
                input[0] = Channel.of( [ [ id:'test', single_end:false ], // meta map
                    file('/Users/sofia/mgnify/assembly-analysis-pipeline/subworkflows/local/rna_annotation/tests/test.fasta.gz', checkIfExists: true)
                ] )
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
