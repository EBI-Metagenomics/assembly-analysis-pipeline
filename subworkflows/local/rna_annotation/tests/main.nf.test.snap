{
    "test_rna_annotation": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_concat_SSU_LSU_coords.txt:md5,2ff422ba8581127bc95262d9751dbcba"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_SSU.fasta:md5,fb23f13526672a300845b767dbc5fd44"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cmsearch_deoverlap:md5,2ff422ba8581127bc95262d9751dbcba"
                    ]
                ],
                "4": [
                    "versions.yml:md5,20bff06ff2925cb1cb36ffbc16f712af",
                    "versions.yml:md5,3e0a8e43b9e83c321a0f944fa5418671",
                    "versions.yml:md5,9eb6b1edf419058844d2df2d1ce6b7b0",
                    "versions.yml:md5,d0acc051be8469d439aa17ba08746157",
                    "versions.yml:md5,d1e5e607889e65c335888aa8dd29a933",
                    "versions.yml:md5,dd49da6d84e60ba329e54acae04d4595"
                ],
                "all_rna_coords": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cmsearch_deoverlap:md5,2ff422ba8581127bc95262d9751dbcba"
                    ]
                ],
                "lsu_fasta": [
                    
                ],
                "ssu_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_SSU.fasta:md5,fb23f13526672a300845b767dbc5fd44"
                    ]
                ],
                "ssu_lsu_coords": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_concat_SSU_LSU_coords.txt:md5,2ff422ba8581127bc95262d9751dbcba"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,20bff06ff2925cb1cb36ffbc16f712af",
                    "versions.yml:md5,3e0a8e43b9e83c321a0f944fa5418671",
                    "versions.yml:md5,9eb6b1edf419058844d2df2d1ce6b7b0",
                    "versions.yml:md5,d0acc051be8469d439aa17ba08746157",
                    "versions.yml:md5,d1e5e607889e65c335888aa8dd29a933",
                    "versions.yml:md5,dd49da6d84e60ba329e54acae04d4595"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-31T16:19:38.08461"
    }
}